test/ledger/surveyor.integration.test.js

```diff
@@ -47,35 +79,45 @@ test('required cohorts are added to surveyors', async (t) => {
   const {
     body: publicSurveyor
   } = await createSurveyor()
-  const { surveyorId: id } = publicSurveyor
+  const { surveyorId } = publicSurveyor
 
   const ledger = await connectToDb('ledger')
   const surveyors = ledger.collection('surveyors', () => {})
-  const privateSurveyor = await querySurveyor('findOne', id)
+  const privateSurveyor = await get()
 
+  const encoded = encodeURIComponent(surveyorId)
   const {
     surveyorType
   } = privateSurveyor
-  const url = `/v2/surveyor/${surveyorType}/${encodeURIComponent(id)}`
+  const url = `/v2/surveyor/${surveyorType}/${encoded}`
 
   await ledgerAgent
     .get(url)
     .expect(ok)
 
-  await timeout(5000)
-
-  const privateFullSurveyor = await querySurveyor('findOne', id)
-  cohorts.map((cohort) => {
+  while (await noCohorts()) {
+    await timeout(5000)
+  }
+  const privateFullSurveyor = await get()
+  cohorts.forEach((cohort) => {
     const surveyorCohortGrants = privateFullSurveyor.cohorts[cohort]
     t.true(_.isArray(surveyorCohortGrants), 'an array is present')
     t.true(surveyorCohortGrants.length > 0, 'the array is not empty')
   })
-  await querySurveyor('remove', id)
-  await postgres.query(`DELETE FROM surveyor_groups WHERE id = $1::text;`, [id])
 
-  function querySurveyor (method, surveyorId) {
-    return surveyors[method]({
+  await surveyors.remove({
+    surveyorId
+  })
+  await postgres.query(`DELETE FROM surveyor_groups WHERE id = $1::text;`, [surveyorId])
+
+  function get () {
```

Comment:
I'm worried this function name could be a bit ambiguous.  Is there a more descriptive name we could use?

Reply:
sure. i can change it to `findOneSurveyor` or something